src/sqlancer/ProviderAdapter.java

```diff
@@ -149,6 +149,9 @@ public void generateAndTestDatabaseWithQueryPlanGuidance(G globalState) throws E
                         mutateTables(globalState);
                         break;
                     }
+                    if (executedQueryCount < globalState.getOptions().getNrQueries()) {
```

Comment:
Perhaps it is more concise to put this checking into line 128 to replace `true`?